test/test.js

```diff
@@ -347,4 +441,227 @@ describe('rollup-plugin-babel', function() {
 			assert.ok(code.indexOf('console.foobaz') !== -1);
 		});
 	});
+
+	it('can be used as an input plugin while transforming the output', () => {
+		return bundle('samples/basic/main.js', {
+			presets: ['@babel/env'],
+			transformGenerated: true,
+		}).then(({ code }) => {
+			assert.ok(code.indexOf('const') === -1, code);
+		});
+	});
+
+	describe('when used as an output plugin', () => {
+		function bundleWithOutputPlugin(input, babelOptions = {}, generateOptions = {}, rollupOptions = {}) {
+			return rollup
+				.rollup(Object.assign({ input }, rollupOptions))
+				.then(bundle => {
+					return bundle.generate(
+						Object.assign(
+							{
+								format: 'cjs',
+								plugins: [babelPlugin(Object.assign({ transformGenerated: true }, babelOptions))],
+							},
+							generateOptions,
+						),
+					);
+				})
+				.then(({ output: [generated] }) => generated);
+		}
+
+		it('allows running the plugin on the output via output options', () => {
+			return bundleWithOutputPlugin('samples/basic/main.js', {
+				presets: ['@babel/env'],
+			}).then(({ code }) => {
+				assert.ok(code.indexOf('const') === -1, code);
+			});
+		});
+
+		it('ignores .babelrc when transforming the output by default', () => {
+			return bundleWithOutputPlugin('samples/basic/main.js').then(({ code }) => {
+				assert.ok(code.indexOf('const') !== -1, code);
+			});
+		});
+
+		it('respects a .babelrc file next to the target when the "file" option is used', () => {
+			return bundle(
+				'samples/babelrc/main.js',
+				{},
+				{
+					file: 'samples/babelrc/bundle.js',
+				},
+			).then(({ code }) => {
+				assert.ok(code.indexOf('const') !== -1, code);
+				assert.ok(code.indexOf('Math.pow') !== -1, code);
+			});
+		});
+
+		it('respects a .babelrc file in the same directory when the "dir" option is used', () => {
```

Comment:
This would be rather confusing for me as a user that using file/dir is supposed to change if we load babel config files. Is loadPartialConfig called conditionally based on that?

Reply:
It is not, but otherwise we cannot pass in a filename, and therefore Babel would not know where to look for configuration files. Using `dir` or `filename` seemed reasonable to me, but if you have better ideas, please go ahead.

We could also say that we completely ignore config files when used on the output, but I am not sure everyone would be happy about that.

Reply:
If you do not pass in `dir` or `filename`, config files will be ignored, but `loadPartialConfig` will still properly resolve plugins and presets based on `process.cwd()`.